tests/Tests/200__monitor_and_manage/200__metrics/200__metrics.robot

```diff
@@ -44,20 +44,30 @@ Test Metric "Notebook CPU Usage" On ODS Prometheus
     Should Not Be Equal    ${cpu_usage_before}    ${cpu_usage_after}
 
 Test Metric "Rhods_Total_Users" On ODS Prometheus
-    [Documentation]    Verifies that metrics rhods_total_users and jupyterhub_total_users have the same value in Prometheus
+    [Documentation]    Verifies that metric value for rhods_total_users
+    ...    matches the value of its corresponding expression
     [Tags]    Sanity
     ...       Tier1
     ...       ODS-628
-    ${expression} =    Set Variable    rhods_total_users&step=1  #step = 1 beacuase it will return value of current second
-    ${rhods_total_users} =    Prometheus.Run Query    ${RHODS_PROMETHEUS_URL}    ${RHODS_PROMETHEUS_TOKEN}    ${expression}
-    Log    rhods_total_users: ${rhods_total_users.json()["data"]["result"][0]["value"][-1]}
-    ${expression} =    Set Variable    jupyterhub_total_users&step=1  #step = 1 beacuase it will return value of current second
-    ${jupyterhub_total_users} =    Prometheus.Run Query    ${RHODS_PROMETHEUS_URL}    ${RHODS_PROMETHEUS_TOKEN}
+
+    # Note: the expression ends with "step=1" to obtain the value for current second
+    ${expression} =    Set Variable    rhods_total_users&step=1
+    ${rhods_total_users} =    Prometheus.Run Query    ${RHODS_PROMETHEUS_URL}    ${RHODS_PROMETHEUS_TOKEN}
+    ...   ${expression}
+    ${rhods_total_users} =    Set Variable    ${rhods_total_users.json()["data"]["result"][0]["value"][-1]}
+    Log    rhods_total_users:${rhods_total_users}
+
+    # Note: the expression ends with "step=1" to obtain the value cor current second
+    ${expression} =    Set Variable
+    ...    count(kube_statefulset_replicas{namespace=~"rhods-notebooks", statefulset=~"jupyter-nb-.*"})&step=1
+    ${total_users_using_expression} =    Prometheus.Run Query    ${RHODS_PROMETHEUS_URL}    ${RHODS_PROMETHEUS_TOKEN}
     ...    ${expression}
-    Log    jupyterhub_total_users: ${jupyterhub_total_users.json()["data"]["result"][0]["value"][-1]}
-    Should Be Equal    ${rhods_total_users.json()["data"]["result"][0]["value"][-1]}
-    ...    ${jupyterhub_total_users.json()["data"]["result"][0]["value"][-1]}
-    ...    msg=metrics rhods_total_users and jupyterhub_total_users don't have the same value
```

Comment:
so there's not metrics replacing the one for JH total users? Like, total users which has used KFNBC or spawned a notebook. sth like that

Reply:
No, kubeflow doesn't provide a metric, as far I know. I was hesitating if to remove this test or improve it checking if the number of users increased after having a new login from a new one, but it's a little tricky because the tests will work first time and not the second time (if we use the same user). Ideas how to implement it are welcomed ...

Reply:
should we ask devs to implement the metric? if necessary

Reply:
The metric rhods_total users is visible in OpenShiftConsole > Observe > Metrics. It's defined [here](https://github.com/red-hat-data-services/odh-deployer/blob/35d0269ee67cefc6a22d2f1b79cd4cac73dff335/monitoring/prometheus/prometheus-configs.yaml#L118) as a recording rule.

This tests is currently just checking that the metric is well defined. As I said, it could be improved in the future to verify that the count increases after new users log-in